src/WebJobs.Script/Host/FunctionMetadataManager.cs

```diff
@@ -89,5 +153,32 @@ internal bool IsScriptFileDetermined(FunctionMetadata functionMetadata)
             }
             return true;
         }
+
+        private void AddMetadataFromCustomProviders(List<FunctionMetadata> functionMetadataList)
+        {
+            var functionProviderTasks = new List<Task<ImmutableArray<FunctionMetadata>>>();
+            foreach (var functionProvider in _functionProviders)
+            {
+                functionProviderTasks.Add(functionProvider.GetFunctionMetadataAsync());
+            }
+
+            var functionMetadataListArray = Task.WhenAll(functionProviderTasks).GetAwaiter().GetResult();
+
+            foreach (var someArray in functionMetadataListArray)
+            {
+                if (!someArray.IsDefaultOrEmpty)
+                {
+                    someArray.ToList().ForEach(el => functionMetadataList.Add(el));
+                }
+            }
+        }
+
+        private void AddErrorsFromCustomProviders()
+        {
+            foreach (var provider in _functionProviders)
+            {
+                provider.FunctionErrors?.ToList().ForEach(kvp => _functionErrors[kvp.Key] = kvp.Value.ToList());
```

Comment:
Seems we need to have a check when aggregating functions from custom providers to prevent multiple functions of the same name from different providers?

Reply:
Yeah good point. I figured that would be handled upstream, but this should be handled here. I added logic to deal with that, and added tests for it. Thanks!